tests/database/test_db.py

```diff
@@ -1,111 +1,112 @@
-            'frequency': 'daily',
-            'url': "http://example-2.com",
-            'schema_type': 'strict',
-            'source_type': 'json'
-    }, str(org_data.id))
-    assert new_source.name == 'Test Source' 
-   
-    sources = db_interface.get_all_harvest_sources()
-    assert any(source['name'] == 'Test Source' for source in sources)
-
-
-def test_add_harvest_job(db_interface):
-    org_data = db_interface.add_organization({'name': 'GSA',
-                'logo': 'url for the logo'})
-    new_source = db_interface.add_harvest_source({
-            'name': 'Test Source',
-            'notification_emails': ['test@example.com'],
-            'frequency': 'daily',
-            'url': "http://example-3.com",
-            'schema_type': 'strict',
-            'source_type': 'json'
-    }, str(org_data.id))
-        
-    job_data = {
-        'status': 'in_progress',
-        'date_created': '2022-01-01',
-        'date_finished': '2022-01-02',
-        'records_added': 10,
-        'records_updated': 5,
-        'records_deleted': 2,
-        'records_errored': 1,
-        'records_ignored': 0
+def app():
+    _app = create_app(testing=True)
+    _app.config['TESTING'] = True
+    _app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///:memory:'
+
+    with _app.app_context():
+        db.create_all()
+        yield _app
+        db.drop_all()
+
+@pytest.fixture(scope='function')
+def session(app):
+    with app.app_context():
+        connection = db.engine.connect()
+        transaction = connection.begin()
+
+        SessionLocal = sessionmaker(bind=connection, autocommit=False, autoflush=False)
+        session = scoped_session(SessionLocal)
+        yield session
+
+        session.remove()
+        transaction.rollback()
+        connection.close()
+
+@pytest.fixture(scope='function')
+def interface(session):
+    return HarvesterDBInterface(session=session)
+
+
+@pytest.fixture
+def org_data():
+    return {'name': 'Test Org', 'logo': 'https://example.com/logo.png'}
+
+@pytest.fixture
+def organization(interface, org_data):
+    org = interface.add_organization(org_data)
+    return org
+
+def test_add_organization(interface, org_data):
+    org = interface.add_organization(org_data)
+    assert org is not None
+    assert org.name == 'Test Org'
+
+def test_get_all_organizations(interface, org_data):
+    interface.add_organization(org_data)
+
+    orgs = interface.get_all_organizations()
+    assert len(orgs) > 0
+    assert orgs[0]['name'] == 'Test Org'
+
+def test_update_organization(interface, organization):
+    updates = {'name': 'Updated Org'}
+    updated_org = interface.update_organization(organization.id, updates)
+    assert updated_org['name'] == 'Updated Org'
+
+def test_delete_organization(interface, organization):
+    result = interface.delete_organization(organization.id)
+    assert result == "Organization deleted successfully"
+
+@pytest.fixture
+def source_data(organization):
```

Comment:
is this where you're consuming the `organization` fixture from L42?

Reply:
yes, correct.